src/bicep/modules/privateLink.bicep

```
@@ -0,0 +1,211 @@
+@description('The name of the resource the private endpoint is being created for')
+param laWorkspaceName string
+
+@description('The resource id of the resoure the private endpoint is being created for')
+param laWorkspaceResourceId string
+
+@description('The region the private endpoint will be created in')
+param privateEndpointLocation string = resourceGroup().location
+
+@description('The name of the subnet in the virtual network where the private endpoint will be placed')
+param privateEndpointSubnetName string
+
+@description('The name of the virtual network where the private endpoint will be placed')
+param privateEndpointVnetName string
+
+@description('The tags that will be associated to the VM')
+param tags object
+
+@description('Data used to append to resources to ensure uniqueness')
+param uniqueData string = substring(newGuid(), 0, 8)
+
+@description('The name of the the resource group where the virtual network exists')
+param vnetResourceGroup string = resourceGroup().name
+
+@description('The subscription id of the subscription the virtual network exists in')
+param vnetSubscriptionId string = subscription().subscriptionId
```

in effort to keep modules unopinionated

```suggestion
@description('The name of the the resource group where the virtual network exists')
param vnetResourceGroupName string

@description('The subscription id of the subscription the virtual network exists in')
param vnetSubscriptionId string
```